### PR TITLE
Drask can no longer eat soap for infinite nutrition

### PR DESCRIPTION
**Pull request by FloFluoro**
## What Does This PR Do
- Drasks now receive 5 nutrition per bite of soap, up from 2.
- Soap is now a finite source of food: After **30** bites, it disappears. The amount of bites remaining can be roughly tracked by examining the soap.
- Splits all soap-eating behavior into its own proc.

## Why It's Good For The Game
Prior to this PR, Drasks could use a single bar of soap as a source of infinite nutrition; the hunger system should not be something that standard players can opt out of with roundstart items. To encourage departmental cooperation, Drasks will now have to periodically consult with the janitor to acquire more food.

## Testing
Compiled and tested on local debug server. The drask had a soap but he eated it

## Changelog
:cl:
tweak: Drask can no longer infinitely eat a single bar of soap. Savor the flavor while it lasts.
tweak: Soap now provides slightly more nutrition to Drasks.
/:cl:

### CODE REVIEW

File: code/game/objects/items/weapons/soap.dm [hunk 1]
```diff
@@ -23,12 +24,27 @@
 	if(target == user && user.a_intent == INTENT_GRAB && ishuman(target))
 		var/mob/living/carbon/human/muncher = user
 		if(muncher && isdrask(muncher))
-			to_chat(user, "<span class='notice'>You take a bite of [src]. Delicious!</span>")
+			times_eaten += 1
 			playsound(user.loc, 'sound/items/eatfood.ogg', 50, 0)
 			user.adjust_nutrition(2)
-		return
+			if(times_eaten <= 3)
```
Comment (farie82) [inline code/game/objects/items/weapons/soap.dm:30]: Try to avoid magic numbers like this. Either make a define or add it as a variable on the item itself.

File: code/game/objects/items/weapons/soap.dm [hunk 2]
```diff
@@ -23,12 +24,27 @@
...
+				to_chat(user, "<span class='notice'>You take a bite of [src]. Delicious!</span>")
+				return
+			else
+				to_chat(user, "<span class='notice'>You finish eating [src].</span>")
+				qdel(src)
+				return
```
Comment (farie82) [inline code/game/objects/items/weapons/soap.dm:36]:
```suggestion
			if(times_eaten <= 3)
				to_chat(user, "<span class='notice'>You take a bite of [src]. Delicious!</span>")
			else
				to_chat(user, "<span class='notice'>You finish eating [src].</span>")
				qdel(src)
			return
```

**Comment by Coolrune206** [comment]
Honestly, while I love the concept of the Drask eating soap meme, some people do manage to take it quite a bit far and do nothing but spam it in front of the bridge for about two hours. Four bites seems a bit small however, I think somewhere around 10 would be a little better.

**Comment by Contrabang** [comment]
We already have infinite and better nutrition sources in the form of mime and clown crayons, which clowns and mimes can just eat forever for 5 rather than 2 nutrition. I think drasks having their own unique thing besides healing a bit when cold is nice, because this effectively makes this tiny bit of fluff useless. Its also not the easiest to acquire said soap in the first place. You're lucky if you find some in maints. If a drask is spamming soap eating in front of the bridge, just steal their soap from them, they rarely have more than one.

File: code/game/objects/items/weapons/soap.dm [hunk 3]
```diff
@@ -23,12 +25,27 @@
 	if(target == user && user.a_intent == INTENT_GRAB && ishuman(target))
 		var/mob/living/carbon/human/muncher = user
 		if(muncher && isdrask(muncher))
-			to_chat(user, "<span class='notice'>You take a bite of [src]. Delicious!</span>")
+			times_eaten += 1
```
Comment (Vi3trice) [inline code/game/objects/items/weapons/soap.dm:28]:
```suggestion
			times_eaten++
```

**Comment by FloFluoro** [comment]
> We already have infinite and better nutrition sources in the form of mime and clown crayons

You are not gonna enjoy my next PR

**Comment by Vermin-the-Varmin** [comment]
Okay so four bites of soap? Is your solution to make it so worthless that all drask can't even eat soap anymore other than as a RP thing

**Comment by Bennyfallout** [comment]
I would suggest removing the nutrient value from the soap rather than just making it disappear after four bites. At the moment it just seems like an incredibly lame thing to do to take away the coolest part about drask other than their cold immunity.

**Comment by Bkulli-VI** [comment]
Its hard to not see this as a spiteful removal  of an aspect of a race, over a mechanic that is probably one of the least impactful things on a shift. to a race that doesn't really have alot of positive upsides or flavor outside of it.

Never has a round been won or lost because someone forgot to eat, much less that a drask was able to eat soap. Nutrition is at most a speed bump that annoys. For any race other than a drask or an IPC, it doesnt add alot of depth, you either go to a vendoring machine, get chicken soup or you buy anything else and try to remember how many tubes of bread you need to eat to not think about it.  Or your other choice is eat whatever the chef made and deal with it, maybe if your lucky botany remember to leave a fruit out on the table.  

RP is likely minimal at the best of times with the chef, and enough vendors drown every department that supplying them is almost never a thought.  So its not like drask are forced out of any kind of solid crew mingling.  In the case of the janitors soap, I would assume more drask would probably prefer to just ask cargo instead. And we all know what that is like.

Its a pretty large flavor that is very important to drask players, Atleast with removing IPC rad immunity it makes them interact with something pretty relevant  even if I do find the logic behind it questionable.  hell there is only like 5 drask players and I  would assume it would be a struggle to think of one of them that would find this kind of a change acceptable.  Again with the IPC case of radiation we got the percentage  damage bonus changed along with it,  From listening on Drask players, they dont really have alot of perks that at good or enjoyable to deal with.

TLDR  this adds little, other then removing a significant flavor to a race to make them engage with a mechanic that is barely a speed bump to most other players.   without much of an uptick to anything else

**Comment by FloFluoro** [comment]
> Is your solution to **make it so worthless** that all drask can't even eat soap anymore other than as a RP thing, Or just you know remove the nutritional value from soap if you think its so game breaking

Soap is still nutritious to Drasks, this PR merely changes it so it can only be eaten a finite amount before being depleted, like every other food in the game.

If it's any consolation, if humans could arbitrarily eat a plate of spaghetti and meatballs an infinite amount of times, I'd want to change that too.

**Comment by GoodApoll0** [comment]
> > Is your solution to **make it so worthless** that all drask can't even eat soap anymore other than as a RP thing, Or just you know remove the nutritional value from soap if you think its so game breaking
> 
> Soap is still nutritious to Drasks, this PR merely changes it so it can only be eaten a finite amount before being depleted, like every other food in the game.
> 
> If it's any consolation, if humans could arbitrarily eat a plate of spaghetti and meatballs an infinite amount of times, I'd want to change that too.

The problem here is that food is replacable. Soap is not that easy to replace. Most rounds as a drask i can't even find soap. Beggings miners and janitors for soap only works so many times. If a human runs out of spaghetti the chef can just make more food or a different type of food. This pr is a good way for there to never be soap in a round cause a drask ate them all.

**Comment by Vermin-the-Varmin** [comment]
> > Is your solution to **make it so worthless** that all drask can't even eat soap anymore other than as a RP thing, Or just you know remove the nutritional value from soap if you think its so game breaking
> 
> Soap is still nutritious to Drasks, this PR merely changes it so it can only be eaten a finite amount before being depleted, like every other food in the game.
> 
> If it's any consolation, if humans could arbitrarily eat a plate of spaghetti and meatballs an infinite amount of times, I'd want to change that too.

My issue is its such a drastic change for such a minor mechanic, no rounds have ever been won or lost due to soap eating, not to mention hunger
And two what about the second point, why make soap which there is a limited ammount of just not give hunger if 2 nutrition per bite is so game breaking and over powered that it needs to be nutered into the ground.
Also don't drasks need to much on it for what five minute to get any meaningful ammount of hunger removed

**Comment by OctusGit** [comment]
Drasks don't really have much going for them. Why nerf the funny and niche feature that gives them character and serves as a minor advantage? Just doesn't seem like an issue when food is only a hot chocolate away.

**Comment by Sirryan2002** [comment]
I like the direction of this change and will put on an approval if the following things are done:
- Increase nutrition gain slightly 
- It would take a lot more than 4 bites to widdle down a bar of soap (its a scarce item), say maybe like 30-35 bites atleast
- Eating Soap too quickly (6-7 bites in the span of 20 seconds?) will have adverse effects, I'm not sure what exactly maybe vomiting, not being able to speak for a minute due to having an extremely soapy mouth, or extreme gain/loss of nutrition 
- Flavor text indicating how much of the soap is left

This will allow drasks to munch on soap for fun and still maintain the fun species unique feature while still addressing the issue of sound bite spam and infinite nutrition taps.

**Comment by FloFluoro** [comment]
> I like the direction of this change and will put on an approval if the following things are done:
> 
> * Increase nutrition gain slightly
> * It would take a lot more than 4 bites to widdle down a bar of soap (its a scarce item), say maybe like 30-35 bites atleast
> * Eating Soap too quickly (6-7 bites in the span of 20 seconds?) will have adverse effects, I'm not sure what exactly maybe vomiting, not being able to speak for a minute due to having an extremely soapy mouth, or extreme gain/loss of nutrition
> * Flavor text indicating how much of the soap is left
> 
> This will allow drasks to munch on soap for fun and still maintain the fun species unique feature while still addressing the issue of sound bite spam and infinite nutrition taps. Soap is probably one of the more rare items in-game, so it being a substitute for actual food is fine since you're giving up a valuable item for a quick meal.

Done.

- Soap can now be bitten 30 times, up from 4
- Soap now adds 5 nutrition, up from 2
- Flavor text gives a rough indicator to how much soap is left
- If soap is eaten 7 or more times in the last 20 seconds, the Drask will keel over and lose most of the nutrition they gained from the soap (see edited OP)

File: code/game/objects/items/weapons/soap.dm [hunk 4]
```diff
@@ -12,6 +12,10 @@
 	throw_range = 20
 	discrete = 1
 	var/cleanspeed = 50 //slower than mop
+	var/times_eaten = 0 //How many times a Drask has chewed on this bar of soap
+	var/max_bites = 30 //The maximum amount of bites before the soap is depleted
+	var/overeat_counter = 0 //How many times this bar of soap has been RECENTLY chewed on.
+	var/overeat_limit = 7 //If you've eaten soap this many times in the last 20 seconds, you get a tummyache
```
Comment (Contrabang) [inline code/game/objects/items/weapons/soap.dm:18]: I think `overeat_limit` might be better as a define, I think `max_bites` is better as a non-define though so that you can adminbus for infinite soap

File: code/game/objects/items/weapons/soap.dm [hunk 5]
```diff
@@ -23,12 +28,47 @@
...
+	. = ..()
+	if(!user.Adjacent(src) || !times_eaten)
+		return
+	if(times_eaten < (max_bites * 0.3))
+		. += "<span class='notice'>[src] has bite marks on it!</span>"
+		return
+	if(times_eaten < (max_bites * 0.6))
+		. += "<span class='notice'>Big chunks of [src] have been chewed off!</span>"
+		return
+	if(times_eaten < (max_bites * 0.9))
```
Comment (Sirryan2002) [inline code/game/objects/items/weapons/soap.dm:67]:
```suggestion
	else if(times_eaten < (max_bites * 0.6))
		. += "<span class='notice'>Big chunks of [src] have been chewed off!</span>"
	else if(times_eaten < (max_bites * 0.9))
```

**Comment by GoodApoll0** [comment]
Why make a downside to eating soap? What's the point in that? Making it so the soap disappears after 30 bites is already a huge downside. Adding a way to lose all nutrition is pointless. Eating soap was just supposed to be a fun mechanic.

**Comment by Vermin-the-Varmin** [comment]
Whats with the stun mechanic, good design should promote good behavior not punish bad ones, why not make it so soap gives less nutrition per bite in that 20 second period
First bite would be max nutrition [a effectively meaningless thing] then the next few bites give less and less till it hits a minimum cap

**Comment by silverplatedelta** [comment]
Not really sure what the point of this PR is or why this is needed. From a balance standpoint I fail to see in anyway why this needs a nerf. Breaking down the drask stat list, the downsides of playing this race are extremely severe. The first downside is the increase in weakness to flashes, which is about the same as vulps (If not the same in the code) leading to flash blinding after two welds without a helm or 5 flashes. Second, drasks take 50% increased burn damage. Given that lasers (and thus burn) is the damage type used by security to lethal antags, drasks are hard countered by sec from shift start, regardless of which antag you rolled, worse if you roll vamp. Even worse if sec gets immolators, its hard crit in 2 hits plus the effect of the next downside. The last official downside is the 200% increased head damage. This is a MASSIVE downside, especially with the viro rework. This translates to an SM delam being lethal in ~15 seconds without a CE/atmos suit, and any virus with a fever being extremely lethal. Another downside is the fact that drask hearts are in their heads. This can lead to both abductors and malpractice bay sending you right to the cloner as you can't be revived for "some reason".

In exchange for all this, drasks get four bonuses of varying significance. The most minor is 5m dark vision, which is nice but common among races. Secondis the cold immunity, which sounds nice but isn't as strong as it sounds. Drasks are able to survive in space for ~15-30 seconds longer due to this and are immune to cryostings and watcher blasts (but not the slowdown, which is the actually dangerous part). Third, drasks get the ability to heal damage when they breath supercooled air. While this sounds extremely overpowered, the healing is only around 0.1-0.5 points of damage every couple seconds. Furthermore, you can only use this perk if someone in atmos/toxins makes a tank for you. 

Finally, the subject of this PR, soap eating. In exchange to being deleted by immolators and quickly dispatched by standard lasers, being blinded by flashes and having doctors not know were their heart is, drasks can opt out of spending 30 seconds going to the kitchen. Often by spending a couple minutes munchin on a bar of soap.

If your issue with this feature is the cronch cronch cronch sounds, then I am all for a change to the sound. It can certainly get annoying. As the previous posters have said, a round has never been "won" or "lost" cause chuthulu munched on some syndie soap instead of eating a grilled cheese. Furthermore, per lore, drasks have a physiological reason for eating soap, their body pH is quite alkaline, and find most food to be gross, instead eating caustic foods, namely soap. 

Not sure what problem this PR is trying to solve. Its not a balance issue as illustrated above. For the argument that its unrealistic: Its a silly spess game, full of zany wacky stuff. We summon demons from slime brains, shoot lasers at a crystal of dark matter, hire known sleeper agents into command roles, and have a goddamn clown and mime on station 24/7. In the grand scheme of things, I don't feel like "its unrealistic" is reason to remove any features from the game.

TLDR: Drasks already got it rough, this is the one major species upside. Why does it need BOTH a nerf and punishment for use, both of which directly conflict with lore.

**Comment by FloFluoro** [comment]
> Text

A species needing buffs is not justification for ignoring design flaws with that species. "Can opt out of the hunger system by chewing on soap" is not meant to be the compensation Drasks get for other maluses. If you think the species is in a bad place, float ideas with the design team on how to improve them.

**Comment by Contrabang** [comment]
>6 times heat damage
I think allowing them to have infinite nutrition like IPCs, and Diona, or literally anyone with a nutritent pump is fine, especially since they have to find soap which is somewhat rare resource

**Comment by Vermin-the-Varmin** [comment]
> > Text
> 
> A species needing buffs is not justification for ignoring design flaws with that species. "Can opt out of the hunger system by chewing on neverending soap" is not meant to be the compensation Drasks get for other maluses. If you think the species is in a bad place, float ideas with the design team on how to improve them.

Are you not going to try and give any suggestions or comments other than a "fuck you i don't care"

**Comment by GoodApoll0** [comment]
> > Text
> 
> A species needing buffs is not justification for ignoring design flaws with that species. "Can opt out of the hunger system by chewing on neverending soap" is not meant to be the compensation Drasks get for other maluses. If you think the species is in a bad place, float ideas with the design team on how to improve them.

Plamamen don't have to eat. Are you going to make a PR that makes them have to eat? Your arguement is very flawed

**Comment by Henri215** [comment]
> > > Text
> > 
> > 
> > A species needing buffs is not justification for ignoring design flaws with that species. "Can opt out of the hunger system by chewing on neverending soap" is not meant to be the compensation Drasks get for other maluses. If you think the species is in a bad place, float ideas with the design team on how to improve them.
> 
> Plamamen don't have to eat. Are you going to make a PR that makes them have to eat? Your arguement is very flawed

Drask specie trait is is that they can eat soap, not "They do not need to eat" boon that plasmamen do have. Lorewise do you think Drasks are creatures in which their culture involves surviving only by eating soap? "they have shown fondness for ice cream, citrus, mint, cucumber, and soap" the characteristic is that they like soap, thats it.

**Comment by Miraviel** [comment]
An infinite resource is getting taken away, it is normal to have such a vocal opposition.

- You can find 10 bars of soap on the Cyberiad at roundstart, not including random spawns.
- You can make an infinite amount of soap with lye and gibs (chemistry).
- You can make an infinite amount of soap with tape rolls and gibs.
- Mining can buy it.
- As a janitor, you start with one.

If your entire drask personality revolves around infinitely eating soap and your character gets nullified by this PR, that is not the PR creator's issue.

>     * Soap can now be bitten 30 times, up from 4
> 
>     * Soap now adds 5 nutrition, up from 2
> 
>     * Flavor text gives a rough indicator to how much soap is left

This is a very good middle-ground, drasks will be still able to monch on soap without doing it for hours while being bridge hobos.

**Comment by silverplatedelta** [comment]
Fair points in response Flo, Henri, and Miraviel. While I agree that the soap shouldn't be the main upside of the species, as things currently stand I would say it is. I can comfortably say that my character's personality doesn't revolve around soap, though it does make for an entertaining bit and open up an opportunity for bribery. 

I think a lot of the opposition is stemming from the lack of a replacement in some sort of perk. Having infinite food was by far the most useful perk of playing a drask, and I feel that it should be given something to replace it. Perhaps a buff to the cold-air- based healing? I'm not the best code diver, but it appears to only heal 0.5 points of burn and 0.25 brute per breath tick (which I think is every couple seconds).

**Comment by GoodApoll0** [comment]
The dislikes on this post clearly show that majority don't want this. If your main reason for wating this is cause of one bridge hobo every few weeks decides to eat soap for a few minutes then that sounds like a you issue not a pr issue.

**Comment by Bkulli-VI** [comment]
I will restate, when IPCs lost rad Immunity. they at least got something out of it in return.

The people that actually use the use the drask find this a fun aspect of them.  and if it isn't logically consistent, or realistic the game drowns in hundreds of other examples.

without something to make it seem like a worthy trade, it comes off alot like someone slapping a fun toy out of someones hand. and leaving it how it is is remarkably low impact on next to anything.  I struggle to see enough players actually caring if you left it alot vs the amount of people losing something they find fun out of the game being removed.

its a major case of doing more harm then help as it stands

**Comment by Contrabang** [comment]
> You can find 10 bars of soap on the Cyberiad at roundstart, not including random spawns.
You can make an infinite amount of soap with lye and gibs (chemistry).
You can make an infinite amount of soap with tape rolls and gibs.
Mining can buy it.
As a janitor, you start with one.

Only about 2, maybe 4 soap you can get without theft or just being that job. Regularly committing theft just because you're a certain race and you want to have species flavor text is a speed run to a bwoink (yes... this has happened, and I think this will make it worse). I've never actually seen anyone use lye/gibs/tape for soap, I know its in the code but every time I've actually tried it its never worked.

And uhh... janitors dont start with soap. Mining can buy it, but its already hard enough for RND to get shit from them, I dont think they're gonna buy anyone soap lol.

Soap isnt actually that common imo

**Comment by silverplatedelta** [comment]
> > > As a former drask only player, I ate soap like every sixth round I had played as Kruum. It was honestly more annoying to spam soap for just the tiniest bit of nutrition.

I just ran some numbers, a buff to 5 nutrition is more than I thought. Id have to guess I eat around 200 points of nutrition a round, on the high end. That's 100 bites of old soap. After the buff, that comes out to 40 bites, or just over a bar and a third, which isn't that egregious. Perhaps we could make a change to the spam stun, up to say 10-12 over seven. It still seems a bit restrictive.

> > You can find 10 bars of soap on the Cyberiad at roundstart, not including random spawns.
> > You can make an infinite amount of soap with lye and gibs (chemistry).
> > You can make an infinite amount of soap with tape rolls and gibs.
> > Mining can buy it.
> > As a janitor, you start with one.
> 
> Only about 2, maybe 4 soap you can get without theft or just being that job. Regularly committing theft just because you're a certain race and you want to have species flavor text is a speed run to a bwoink (yes... this has happened, and I think this will make it worse). I've never actually seen anyone use lye/gibs/tape for soap, I know its in the code but every time I've actually tried it its never worked.
> 
> And uhh... janitors dont start with soap. Mining can buy it, but its already hard enough for RND to get shit from them, I dont think they're gonna buy anyone soap lol.

In regards to this matter, I think that soap being harder to get can encourage RP, and I often use soap hunting as something to keep myself occupied early in the round as an officer/command member. 

Something else to just consider, its consumption will have consequences, notably on cult rounds when officers go looking for soap. Just something to keep in mind.

**Comment by Denthamos** [comment]
Anything that provides infinite nutrition will not have a place in the future here if we want the chef and botanist roles to be more important with what they perform for the station.

> - If a Drask eats soap too many times in a short period of time (7 or more times in 20 seconds), the soap's lye irritates their stomach, causing them to keel over for awhile and lose most of the nutrition they gained.

This part seems a bit...egh?  Like why should they be punished for eating something safe for them to eat when you can also spam eat low nutrition food with no downsides?  Remove that part and I will approve it as well, otherwise denied on my end.

As for adding another feature to drask I would be all for them getting another little flavor feature as long as it is not heavily impacting on their balance or how they play the round as a whole but that would be another PR entirely and out of scope for this one.

**Comment by FloFluoro** [comment]
>This part seems a bit...egh? Like why should they be punished for eating something safe for them to eat when you can also spam eat low nutrition food with no downsides? Remove that part and I will approve it as well, otherwise denied on my end.

This feature was requested by @Sirryan2002 as a prerequisite to _their_ design approval, you might want to talk with them. I'd be fine rooting it out or tweaking it as needed.

**Comment by GoodApoll0** [comment]
> Anything that provides infinite nutrition will not have a place in the future here if we want the chef and botanist roles to be more important with what they perform for the station.
> 
> > * If a Drask eats soap too many times in a short period of time (7 or more times in 20 seconds), the soap's lye irritates their stomach, causing them to keel over for awhile and lose most of the nutrition they gained.
> 
> This part seems a bit...egh? Like why should they be punished for eating something safe for them to eat when you can also spam eat low nutrition food with no downsides? Remove that part and I will approve it as well, otherwise denied on my end. @FloFluoro
> 
> As for adding another feature to drask I would be all for them getting another little flavor feature as long as it is not heavily impacting on their balance or how they play the round as a whole but that would be another PR entirely and out of scope for this one.
> 
> Also as a note - you can order soap in from cargo from a janitorial crate using your own money with the economy system now as a drask.

Do you think a buff to the cooled air healing would be good? I feel it's a too low as it is now. even if it's just 1burn/brute per tick would suffice

**Comment by Vermin-the-Varmin** [comment]
> > This part seems a bit...egh? Like why should they be punished for eating something safe for them to eat when you can also spam eat low nutrition food with no downsides? Remove that part and I will approve it as well, otherwise denied on my end.
> 
> This feature was requested by @Sirryan2002 as a prerequisite to their design approval, you might want to talk with them. I'd be fine rooting it out or tweaking it as needed.

Just wondering why that wasn't stated in the original PR message, keep everyone in the dark that the most hated part of the PR was a required part of the PR

**Comment by Henri215** [comment]
> > > This part seems a bit...egh? Like why should they be punished for eating something safe for them to eat when you can also spam eat low nutrition food with no downsides? Remove that part and I will approve it as well, otherwise denied on my end.
> > 
> > 
> > This feature was requested by @Sirryan2002 as a prerequisite to their design approval, you might want to talk with them. I'd be fine rooting it out or tweaking it as needed.
> 
> Just wondering why that wasn't stated in the original PR message, keep everyone in the dark that the most hated part of the PR was a required part of the PR

What do you mean by keeping in the dark? Sirryan comment is right there, its public.

**Comment by Sirryan2002** [comment]
I'll address a few things,
1) I had no prior interaction with this PR before leaving my initial comment, so don't accuse me or the author of hiding things
2) If Denth wants the maluse of eating soap removed, that's perfectly fine with me, you can remove the weaken and nutrition loss, it would be funny if something funny happened like their mouth soaping up or something. I just want their to be some form of things preventing people from sound byte spamming

**Comment by hal9000PR** [comment]
I would be happy for someone to make a PR to buff drask in a way, eg their super cooled air healing. 

But that's out of the scope of this pr.

I really shouldn't have to say this but do **not** attack people for their ideas. You can debate ideas but hostilities have no place here.

**Comment by GDNgit** [comment]
I like the compromise in this PR, however I do have one (minor) request. It may be good to log when a bar of soap is finished being eaten. Just to make sure that someone isn't going around the station eating every bar of soap to be a jackass.

**Comment by GoodApoll0** [comment]
If the biggest issue is sound spam, i think a cooldown on the cronch sound would be useful? Like how you can't emote spam without having to wait.

**Review by farie82 — APPROVED** [review]
